src/AssociatedJobInfoClient.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.DotNet.HelixPoolProvider.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.HelixPoolProvider
+{
+    public class AssociatedJobInfo
+    {
+        public string BuildSourceBranch { get; }
+        public string SystemPullRequestTargetBranch { get; }
+
+        public static readonly AssociatedJobInfo Empty
+            = new AssociatedJobInfo("", "");
+
+        public AssociatedJobInfo(
+            string buildSourceBranch,
+            string systemPullRequestTargetBranch)
+        {
+            BuildSourceBranch = buildSourceBranch;
```

Comment:
Out of curiosity, why is anything other than the target branch needed for this?

Reply:
I did it in similar way as in Helix SDK. If it is a rolling build we can use `Build.SourceBranch` variable. If it's a PR then `Build.SourceBranch` is set to PR-related branch (e.g. `refs/pull/123/merge`) and because of that we have to also read `System.PullRequest.TargetBranch`. The `Build.SourceBranch` is appended to the `Job.Source` string, the other one is put into `Job.Properties`.